src/api/telemetry/TelemetryCollection.js

```diff
@@ -307,8 +312,16 @@ export class TelemetryCollection extends EventEmitter {
      * @private
      */
     _timeSystem(timeSystem) {
-        this.timeKey = timeSystem.key;
-        let metadataValue = this.metadata.value(this.timeKey) || { format: this.timeKey };
+        let domains = this.metadata.valuesForHints(['domain']);
+        let domain = domains.filter((d) => d.key === timeSystem.key)[0];
```

Comment:
This is exactly what [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) is for. It has the additional benefit of short-circuiting when it finds a matching element.